src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java

```diff
@@ -99,18 +107,47 @@ public String toString() {
     }
 
     /**
+     * Creates a new FullIdent starting from the child of the specified node.
+     * FullIdent can include array.
+     * @param ast the parent node from where to start from
+     * @param isArrayDeclare the FullIdent will be include array
+     * @return a {@code FullIdent} value
+     */
+    private static FullIdent createFullIdentMaybeArrayDeclare(DetailAST ast,
+                                                              boolean isArrayDeclare) {
+        final FullIdent ident = new FullIdent();
+        extractFullIdent(ident, ast, isArrayDeclare);
+        return ident;
+    }
+
+    /**
      * Recursively extract a FullIdent.
      *
      * @param full the FullIdent to add to
      * @param ast the node to recurse from
+     * @param isArrayDeclare the FullIdent will be include array
      */
-    private static void extractFullIdent(FullIdent full, DetailAST ast) {
+    private static void extractFullIdent(FullIdent full, DetailAST ast,
+                                         boolean isArrayDeclare) {
```

Comment:
Why do we need `isArrayDeclare` and all those additional methods like `createFullIdentMaybeArrayDeclare` and `createFullIdentMaybeArrayDeclare`? Do we even need extracting FullIdent without arrays?

Reply:
If I change old `extractFullIdent` on new, then next test crashed:
![new bitmap image](https://user-images.githubusercontent.com/19771549/30158465-173ddf12-93ce-11e7-9e32-155e9dc7d4f5.png)
[testRegularExpression](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java#L178)
[testEmptyRegularExpression](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java#L194)
[test](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java#L42)
[testCreateFullTypeOfArray](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java#L42)
Of course, I can rewrite the tests, but I don't think this is the best solution.

Reply:
I think here https://github.com/checkstyle/checkstyle/blob/66cc9d43c9620678078d2716308ee61916227fcb/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java#L213
we just need to use https://github.com/checkstyle/checkstyle/blob/1d9b547ac98d69166db532a7cb275080206ff097/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtils.java#L159
that has to resolve problems of `ClassFanOutComplexityCheck`

Reply:
@Vladlis , fixed `createFullIdent`. But, I can't extract change from `FullIdent ` to `CheckUtils`, because I can use method `append `  only inside `FullIdent`. I can't change private on public, because `FullIdent `this is API.